nixos/modules/services/mail/zeyple.nix

```
@@ -0,0 +1,91 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.services.zeyple;
+  bool2int = x: if x then "1" else "0";
+
+  gpgHome = pkgs.runCommand "zeyple-gpg-home" { } ''
+    mkdir -p $out
+    for file in ${lib.concatStringsSep " " cfg.keys}; do
+      ${pkgs.gnupg}/bin/gpg --homedir="$out" --import "$file"
+    done
+
+    # Remove socket files
+    rm -f $out/S.gpg-agent*
+  '';
+in {
+  options.services.zeyple = {
+    enable = mkEnableOption "Zeyple, an utility program to automatically encrypt outgoing emails with GPG";
+
+    user = mkOption {
+      type = types.str;
+      default = "zeyple";
+      description = "User to run Zeyple as.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "zeyple";
+      description = "Group to use to run Zeyple.";
+    };
+
+    logFile = mkOption {
+      type = types.path;
+      default = "/var/log/zeyple.log";
+      description = "Path of the log file.";
+    };
+
+    forceEncrypt = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Whether to abort sending mail if no keys are found.";
+    };
```

In my opinion it might be worth replacing with a single `settings` option...

```suggestion
    settings = mkOption {
      type = (pkgs.formats.ini {}).type;
      default = { };
      description = ''
        Zeyple configuration. refer to
        <link xlink:href="https://github.com/infertux/zeyple/blob/master/zeyple/zeyple.conf.example"/>
        for details on supported values.
      '';
    };

    forceEncrypt = mkOption {
      type = types.bool;
      default = true;
      description = "Whether to abort sending mail if no keys are found.";
    };
```